test/suites/integration/user/file-upload-storage-request.test.ts

```diff
@@ -0,0 +1,466 @@
+import "@storagehub/api-augment";
+import { strictEqual } from "node:assert";
+import { after, before, describe, it } from "node:test";
+import {
+  DUMMY_MSP_ID,
+  NODE_INFOS,
+  TEST_ARTEFACTS,
+  createApiObject,
+  fetchEventData,
+  runBspNet,
+  shUser,
+  checkBspForFile,
+  checkFileChecksum,
+  type BspNetApi,
+  type BspNetConfig,
+  closeBspNet,
+  sleep
+} from "../../../util";
+import { GenericAccountId32 } from "@polkadot/types";
+
+const bspNetConfigCases: BspNetConfig[] = [
+  { noisy: false, rocksdb: false },
+  { noisy: false, rocksdb: true }
+];
+
+for (const bspNetConfig of bspNetConfigCases) {
+  describe("User: Issue Storage Requests", () => {
+    let user_api: BspNetApi;
+    let bsp_api: BspNetApi;
+
+    before(async () => {
+      await runBspNet(bspNetConfig);
+      user_api = await createApiObject(`ws://127.0.0.1:${NODE_INFOS.user.port}`);
+      bsp_api = await createApiObject(`ws://127.0.0.1:${NODE_INFOS.bsp.port}`);
+    });
+
+    after(async () => {
+      await user_api.disconnect();
+      await bsp_api.disconnect();
+      await closeBspNet();
+    });
+    it("issueStorageRequest sent correctly", async () => {
+      const destination = "test/smile.jpg";
+      const bucketName = "bucket-2";
+
+      const newBucketEventEvent = await user_api.createBucket(bucketName);
+      const newBucketEventDataBlob =
+        user_api.events.fileSystem.NewBucket.is(newBucketEventEvent) && newBucketEventEvent.data;
+
+      if (!newBucketEventDataBlob) {
+        throw new Error("Event doesn't match Type");
+      }
+
+      const issueStorageRequestResult = await user_api.sealBlock(
+        user_api.tx.fileSystem.issueStorageRequest(
+          newBucketEventDataBlob.bucketId,
+          destination,
+          TEST_ARTEFACTS["res/smile.jpg"].fingerprint,
+          TEST_ARTEFACTS["res/smile.jpg"].size,
+          DUMMY_MSP_ID,
+          [NODE_INFOS.user.expectedPeerId]
+        ),
+        shUser
+      );
+
+      // wait for the bsp to volunteer
+      await sleep(500);
+
+      const { event } = user_api.assertEvent(
+        "fileSystem",
+        "NewStorageRequest",
+        issueStorageRequestResult.events
+      );
+
+      const dataBlob = user_api.events.fileSystem.NewStorageRequest.is(event) && event.data;
+
+      if (!dataBlob) {
+        throw new Error("Event doesn't match Type");
+      }
+
+      strictEqual(dataBlob.who.toString(), NODE_INFOS.user.AddressId);
+      strictEqual(dataBlob.location.toHuman(), destination);
+      strictEqual(dataBlob.fingerprint.toString(), TEST_ARTEFACTS["res/smile.jpg"].fingerprint);
+      strictEqual(dataBlob.size_.toBigInt(), TEST_ARTEFACTS["res/smile.jpg"].size);
+      strictEqual(dataBlob.peerIds.length, 1);
+      strictEqual(dataBlob.peerIds[0].toHuman(), NODE_INFOS.user.expectedPeerId);
+    });
```

Comment:
This test is repeated in the BSPs testsuit? Why is it needed here too?

Reply:
Actually I missed removing it, I will delete it from the user tests because they are already extensive and leave it on the BSP side.